### PR TITLE
Make deck proxy build logs for Jenkins jobs.

### PR DESCRIPTION
**Pull request by rmmh**
It needs to be tested, but this doesn't look too gross.

### CODE REVIEW

**Comment by spxtr** [comment]
Spiffy. I'll review tomorrow.

**Comment by rmmh** [comment]
It would probably be best to make a test deployment to exercise the alternate code path.

And it probably should only give a log link if the build has started-- similar to how a pod name is not assigned until it's started.

File: prow/cmd/deck/jobs.go [hunk 1]
```diff
@@ -79,6 +83,33 @@ func (ja *JobAgent) Jobs() []Job {
...
+		if m == nil {
+			return nil, fmt.Errorf("GetLog invalid job name %s", name)
+		}
+		number, err := strconv.Atoi(m[1])
+		if err != nil {
+			return nil, err
+		}
+		return ja.jc.GetLog(m[0], number)
+	}
+	return nil, fmt.Errorf("Cannot get log for %s", name)
```
Comment (spxtr) [inline prow/cmd/deck/jobs.go]: Errors start with lowercase in go.

**Comment by spxtr** [comment]
This looks reasonable to me. I can help set up a test deployment.

**Comment by rmmh** [comment]
@k8s-bot bazel test this

**Comment by rmmh** [comment]
@spxtr PTAL the line parts of this need to be deployed before the deck changes can be tested.

File: prow/cmd/line/main.go [hunk 2]
```diff
@@ -228,6 +228,7 @@ func fields(c *testClient) logrus.Fields {
 // passed in as environment variables. We also include the service account
 // secret.
 func (c *testClient) TestKubernetes() error {
+	agent := "kubernetes"
```
Comment (spxtr) [inline prow/cmd/line/main.go]: Put these in a global constant somewhere.

File: prow/line/line.go [hunk 3]
```diff
@@ -334,7 +334,7 @@ func deleteKubeJob(k deleteClient, job kube.Job) error {
 	return nil
 }
 
-func SetJobStatus(k *kube.Client, podName, jobName, state, desc, url string) error {
+func SetJobStatus(k *kube.Client, podName, jobName, state, desc, url, agent string) error {
```
Comment (spxtr) [inline prow/line/line.go]: We can set agent when the job is created, so that it doesn't need to be reset to the same value on every status change.

**Comment by rmmh** [comment]
Longer than things stay on the prow dashboard. About a week?

**Comment by ixdy** [comment]
yep, [7 days](https://github.com/kubernetes/test-infra/blob/master/jenkins/job-configs/kubernetes-jenkins-pull/bootstrap-pull.yaml#L5-L6).

File: prow/line/line.go (hunk 3 above)
Comment (rmmh) [inline prow/line/line.go]: Deck doesn't know what agent a job will be run on. That's a good idea for a future refactoring.

Comment (spxtr) [inline prow/line/line.go]: Deck doesn't start the job, though. We can change the `line.Start*Job` functions to accept the agent string and set it once.

**Comment by rmmh** [comment]
@spxtr cmd/line/main.go:main() is the one place that currently does the agent dispatching based on whether the presubmit spec exists or not. This logic could be extracted or duplicated if you want, but it seems fine to leave it here for now.

I refactored it to set the agent annotation separately, so the calls to tryCreateStatus aren't changed anymore.

**Pull request merged**